dsgrid/config/file_schemas.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from pathlib import Path
+from typing import Self
+
+from pydantic import Field, field_validator, model_validator
+
+from dsgrid.data_models import DSGBaseModel
+from dsgrid.dimension.base_models import DimensionType
+from dsgrid.exceptions import DSGInvalidDataset, DSGInvalidField
+from dsgrid.spark.functions import read_csv_duckdb, read_json, read_parquet
+from dsgrid.spark.types import DataFrame
+from dsgrid.utils.utilities import check_uniqueness
+
+
+SUPPORTED_TYPES = set(
+    (
+        "BOOLEAN",
+        "INT",
+        "INTEGER",
+        "TINYINT",
+        "SMALLINT",
+        "BIGINT",
+        "FLOAT",
+        "DOUBLE",
+        "TIMESTAMP_TZ",
+        "TIMESTAMP_NTZ",
+        "STRING",
+        "TEXT",
+        "VARCHAR",
+    )
+)
+
+DUCKDB_COLUMN_TYPES = {
+    "BOOLEAN": "BOOLEAN",
+    "INT": "INTEGER",
+    "INTEGER": "INTEGER",
+    "TINYINT": "TINYINT",
+    "SMALLINT": "INTEGER",
+    "BIGINT": "BIGINT",
+    "FLOAT": "FLOAT",
+    "DOUBLE": "DOUBLE",
+    "TIMESTAMP_TZ": "TIMESTAMP WITH TIME ZONE",
+    "TIMESTAMP_NTZ": "TIMESTAMP",
+    "STRING": "VARCHAR",
+    "TEXT": "VARCHAR",
+    "VARCHAR": "VARCHAR",
+}
+
+SPARK_COLUMN_TYPES = {
+    "BOOLEAN": "BOOLEAN",
+    "INT": "INT",
+    "INTEGER": "INT",
+    "TINYINT": "TINYINT",
+    "SMALLINT": "SMALLINT",
+    "BIGINT": "BIGINT",
+    "FLOAT": "FLOAT",
+    "DOUBLE": "DOUBLE",
+    "STRING": "STRING",
+    "TEXT": "STRING",
+    "VARCHAR": "STRING",
+    "TIMESTAMP_TZ": "TIMESTAMP",
+    "TIMESTAMP_NTZ": "TIMESTAMP_NTZ",
+}
+
+assert sorted(DUCKDB_COLUMN_TYPES.keys()) == sorted(SPARK_COLUMN_TYPES.keys())
+assert not SUPPORTED_TYPES.difference(DUCKDB_COLUMN_TYPES.keys())
+
+logger = logging.getLogger(__name__)
+
+
+class Column(DSGBaseModel):
+    name: str = Field(description="Name of the column")
+    dimension_type: DimensionType | None = Field(
+        default=None,
+        description="Dimension represented by the data in the column. Optional if this is a "
+        "time column or pivoted column. Required if the column represents a stacked dimension "
+        "but an alternate name is being used, such as 'county' instead of 'geography'. "
+        "dsgrid will rename any column that is set at runtime.",
+    )
+    data_type: str | None = Field(
+        description="Type of the data in the column. If None, infer the type."
+    )
+
+    @field_validator("data_type")
+    @classmethod
+    def check_data_type(cls, data_type: str | None) -> str | None:
+        if data_type is None:
+            return None
+
+        type_upper = data_type.upper()
+        if type_upper not in SUPPORTED_TYPES:
+            supported_data_types = sorted(SUPPORTED_TYPES)
+            msg = f"{data_type=} is not one of {supported_data_types=}"
+            raise ValueError(msg)
+        return type_upper
+
+
+class FileSchema(DSGBaseModel):
+    """Defines the format of a data file (CSV, JSON, Parquet)."""
+
+    path: str
```

Comment:
Perhaps this should be optional. It is a bit odd to require a data file in a project repository when that data file is not part of the repository.